app/addons/documents/views.js

```diff
@@ -453,25 +458,31 @@ function(app, FauxtonAPI, Components, Documents, Databases, Views, QueryOptions)
       }
 
       this.toggleTrash();
+      this.updateExpandButtonLabel();
     },
 
     serialize: function() {
       return {
         viewList: this.viewList,
-        expandDocs: this.expandDocs,
         endOfResults: !this.pagination.canShowNextfn()
       };
     },
 
-    collapse: function (event) {
-      event.preventDefault();
+    toggleExpandCollapse: function (e) {
+      e.preventDefault();
+      var expand = this.$(e.target).find("i").hasClass('icon-plus');
 
-      if (this.expandDocs) {
-        this.expandDocs = false;
```

Comment:
Yeah... I kinda agree. I'll change this.

One thing though: I never liked the simple toggling of a boolean to keep track of something - I've found it can lead to problems when there's disconnect between what's in the var, and the reality of what's actually in the DOM. But I guess this'll be eliminated once we're using react state, which will have a 1-1 relation.